### PR TITLE
[SOS] [Linux] Initial support of reading source and line number by native offset

### PR DESCRIPTION
**Pull request by lucenticus**
This pull request enables reading source name and line numbers from portable pdb by native offset.
 Now sos commands which are using function `GetLineByOffset` (i.e. `clru, clrstack, ip2md, pe`) show line numbers and source name. This pull request is related with `System.Diagnostics.Debug.SymbolReader.dll` from corefx PR https://github.com/dotnet/corefx/pull/9571
Related issue: https://github.com/dotnet/coreclr/issues/5920
CC @Dmitri-Botcharnikov @chunseoklee @seanshpark @mikem8361

### CODE REVIEW

File: src/ToolBox/SOS/Strike/util.cpp [hunk 1]
```diff
@@ -6291,6 +6295,40 @@ HRESULT SymbolReader::LoadCoreCLR()
...
+
+    if (getLineByILOffsetDelegate == nullptr)
+    {
+        Status = SymbolReader::LoadCoreCLR();
+    }
+    if (Status != S_OK)
+    {
+        return Status;
+    }
+    BSTR wszFileName = SysAllocStringLen(0, cbFileName);
```
Comment (adityamandaleeka) [inline src/ToolBox/SOS/Strike/util.cpp]: Check for null?

File: src/ToolBox/SOS/Strike/util.cpp [hunk 2]
```diff
@@ -6674,6 +6712,30 @@ WString GetFrameFromAddress(TADDR frameAddr, IXCLRDataStackWalk *pStackWalk, BOO
     return frameOutput;
 }
 
+String ModuleNameFromIP(CLRDATA_ADDRESS ip)
+{
+    CLRDATA_ADDRESS mdesc = 0;
+    if (SUCCEEDED(g_sos->GetMethodDescPtrFromIP(ip, &mdesc)))
+    {
+        DacpMethodDescData mdescData;
+        DacpModuleData dmd;
```
Comment (adityamandaleeka) [inline src/ToolBox/SOS/Strike/util.cpp]: How about 'moduleData' instead of 'dmd'?

File: src/ToolBox/SOS/Strike/util.cpp [hunk 3]
```diff
@@ -5900,7 +5902,10 @@ GetLineByOffset(
         }
 
         IlOffset = symInfo.Offset + MethodOffs;
-
+#else
+        const String &moduleName = ModuleNameFromIP(TO_CDADDR(Offset));
+        return SymbolReader::GetLineByILOffset(moduleName.c_str(), MethodToken, MethodOffs, pLinenum, lpszFileName, cbFileName);
```
Comment (adityamandaleeka) [inline src/ToolBox/SOS/Strike/util.cpp]: Does the FEEFEE stuff not apply to this scenario?

Comment (mikem8361) [inline src/ToolBox/SOS/Strike/util.cpp]: Hopefully the FEEFEE sequence points are“filtered” out on the managed side. I don’t remember if it was though.

**Comment by lucenticus** [comment]
@adityamandaleeka I've fixed code, thanks for review! Please note that I've added 0xFEEFEE processing  in the commit https://github.com/dotnet/corefx/pull/9571/commits/a5972e807612c49a61b4039f8100f3932eaf4883

**Comment by mikem8361** [comment]
LGTM. Is this ready to be merged?

**Comment by lucenticus** [comment]
I fixed all comments, you can merge it if @adityamandaleeka has no objection

**Comment by adityamandaleeka** [comment]
LGTM, thanks for updating.

**Pull request merged**